### PR TITLE
EnGoma OK

### PR DESCRIPTION
**Pull request by Synray**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 1]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    ICHAIN_F32(targetArrowOffset, 20, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Goma/func_80A4B554.s")
+void EnGoma_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGoma* this = THIS;
+    s16 params;
+
+    this->eggTimer = Rand_ZeroOne() * 200.0f;
+    Actor_ProcessInitChain(thisx, sInitChain);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:123]:
```suggestion
    Actor_ProcessInitChain(&this->actor, sInitChain);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 2]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    Actor_SetScale(thisx, 0.01f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:124]:
```suggestion
    Actor_SetScale(&this->actor, 0.01f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 3]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    params = this->actor.params;
+
+    if (params >= 100) { // piece of boss goma
+        Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, thisx, ACTORCAT_BOSS);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:128]:
```suggestion
        Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_BOSS);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 4]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+            this->spawnNum = sSpawnNum++;
+        } else if (thisx->params == 9 || thisx->params == 7) {
+            this->actionFunc = EnGoma_Egg;
+        }
+
+        if (thisx->params >= 8) { // on ceiling
+            this->eggYOffset = -1500.0f;
+        } else {
+            this->eggYOffset = 1500.0f;
+        }
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:169]:
```suggestion
        if (this->actor->params < 3) { // Spawned by boss
            this->actionFunc = EnGoma_EggFallToGround;
            this->invincibilityTimer = 10;
            this->actor.speedXZ = 1.5f;
        } else if (this->actor.params == 8 || this->actor.params == 6) {
            this->actionFunc = EnGoma_Egg;
            this->spawnNum = sSpawnNum++;
        } else if (&this->actor.params == 9 || this->actor.params == 7) {
            this->actionFunc = EnGoma_Egg;
        }

        if (thisx->params >= 8) { // on ceiling
            this->eggYOffset = -1500.0f;
        } else {
            this->eggYOffset = 1500.0f;
        }
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 5]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+
+        this->gomaType = ENGOMA_EGG;
+        this->eggScale = 1.0f;
+        this->eggSquishAngle = Rand_ZeroOne() * 1000.0f;
+        this->actionTimer = 50;
+        Collider_InitCylinder(globalCtx, &this->colCyl1);
+        Collider_SetCylinder(globalCtx, &this->colCyl1, thisx, &D_80A4B7A0);
+        Collider_InitCylinder(globalCtx, &this->colCyl2);
+        Collider_SetCylinder(globalCtx, &this->colCyl2, thisx, &D_80A4B7CC);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:178]:
```suggestion
        Collider_SetCylinder(globalCtx, &this->colCyl1, &this->actor, &D_80A4B7A0);
        Collider_InitCylinder(globalCtx, &this->colCyl2);
        Collider_SetCylinder(globalCtx, &this->colCyl2, &this->actor, &D_80A4B7CC);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 6]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    EnGoma* this = THIS;
+
+    if (this->actor.params < 10) {
+        Collider_DestroyCylinder(globalCtx, &this->colCyl1);
+        Collider_DestroyCylinder(globalCtx, &this->colCyl2);
+    }
+}
+
+void EnGoma_SetupFlee(EnGoma* this) {
+    Animation_Change(&this->skelanime, &D_06003D78, 2.0f, 0.0f, Animation_GetLastFrame(&D_06003D78), 0, -2.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:192]:
```suggestion
    Animation_Change(&this->skelanime, &D_06003D78, 2.0f, 0.0f, Animation_GetLastFrame(&D_06003D78), ANIMMODE_LOOP, -2.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 7]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+        this->playerDetectionTimer = 0;
+    }
+
+    if (!(this->eggTimer & 0xF) && Rand_ZeroOne() < 0.5f) {
+        for (i = 0; i < 2; i++) {
+            Vec3f vel = { 0.0f, 0.0f, 0.0f };
+            Vec3f acc = { 0.0f, -0.5f, 0.0f };
+            Vec3f pos;
+
+            // lol poop
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:297]: Im assuming this wasn't meant to be left in lol.

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 8]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+            pos.x = Rand_CenteredFloat(30.0f) + this->actor.world.pos.x;
+            pos.y = Rand_ZeroFloat(30.0f) + this->actor.world.pos.y;
+            pos.z = Rand_CenteredFloat(30.0f) + this->actor.world.pos.z;
+            EffectSsHahen_Spawn(globalCtx, &pos, &vel, &acc, 0, (s16)(Rand_ZeroOne() * 5.0f) + 10, -1, 10, 0);
+        }
+    }
+}
+
+void EnGoma_SetupHatch(EnGoma* this, GlobalContext* globalCtx) {
+    Animation_Change(&this->skelanime, &D_06000544, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000544), 2, 0.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:307]:
```suggestion
    Animation_Change(&this->skelanime, &D_06000544, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000544), ANIMMODE_ONCE, 0.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 9]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+
+void EnGoma_Hatch(EnGoma* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelanime);
+    if (this->actionTimer == 0) {
+        EnGoma_SetupStand(this);
+    }
+}
+
+void EnGoma_SetupHurt(EnGoma* this, GlobalContext* globalCtx) {
+    Animation_Change(&this->skelanime, &D_06000838, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000838), 2, -2.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:327]:
```suggestion
    Animation_Change(&this->skelanime, &D_06000838, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000838), ANIMMODE_ONCE, -2.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 10]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+        if ((s8)this->actor.colChkInfo.health <= 0) {
+            EnGoma_SetupDie(this);
+        } else {
+            EnGoma_SetupFlee(this);
+        }
+    }
+}
+
+void EnGoma_SetupDie(EnGoma* this) {
+    Animation_Change(&this->skelanime, &D_06000B78, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000B78), 2, -2.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:363]:
```suggestion
    Animation_Change(&this->skelanime, &D_06000B78, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000B78), ANIMMODE_ONCE, -2.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 11]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+        }
+    }
+
+    if (this->actionTimer == 0) {
+        EnGoma_SetupDead(this);
+    }
+}
+
+void EnGoma_SetupDead(EnGoma* this) {
+    Animation_Change(&this->skelanime, &D_06000334, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000334), 0, -2.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:398]:
```suggestion
    Animation_Change(&this->skelanime, &D_06000334, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000334), ANIMMODE_LOOP, -2.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 12]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    }
+    this->visualState = 2;
+}
+
+void EnGoma_SetupStand(EnGoma* this) {
+    f32 lastFrame;
+
+    lastFrame = Animation_GetLastFrame(&D_06001548);
+    this->actionTimer = Rand_S16Offset(10, 30);
+    Animation_Change(&this->skelanime, &D_06001548, 1.0f, 0.0f, lastFrame, 0, -5.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:436]:
```suggestion
    Animation_Change(&this->skelanime, &D_06001548, 1.0f, 0.0f, lastFrame, ANIMMODE_LOOP, -5.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 13]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    this->actionFunc = &EnGoma_Stand;
+    this->gomaType = ENGOMA_NORMAL;
+}
+
+void EnGoma_SetupChasePlayer(EnGoma* this) {
+    Animation_Change(&this->skelanime, &D_06003D78, 1.0f, 0.0f, Animation_GetLastFrame(&D_06003D78), 0, -5.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:442]:
```suggestion
    Animation_Change(&this->skelanime, &D_06003D78, 1.0f, 0.0f, Animation_GetLastFrame(&D_06003D78), ANIMMODE_LOOP, -5.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 14]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    this->actionFunc = &EnGoma_ChasePlayer;
+    this->actionTimer = Rand_S16Offset(70, 110);
+}
+
+void EnGoma_SetupPrepareJump(EnGoma* this) {
+    Animation_Change(&this->skelanime, &D_06000E4C, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000E4C), 2, -5.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:448]:
```suggestion
    Animation_Change(&this->skelanime, &D_06000E4C, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000E4C), ANIMMODE_ONCE, -5.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 15]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    Math_ApproachS(&this->actor.shape.rot.y, targetAngle, 2, 3000);
+
+    if (this->actionTimer == 0) {
+        EnGoma_SetupJump(this);
+    }
+    this->visualState = 0;
+}
+
+void EnGoma_SetupLand(EnGoma* this) {
+    Animation_Change(&this->skelanime, &D_0600017C, 1.0f, 0.0f, Animation_GetLastFrame(&D_0600017C), 2, 0.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:470]:
```suggestion
    Animation_Change(&this->skelanime, &D_0600017C, 1.0f, 0.0f, Animation_GetLastFrame(&D_0600017C), ANIMMODE_ONCE, 0.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 16]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    this->actionFunc = &EnGoma_Land;
+    this->actionTimer = 10;
+}
+
+void EnGoma_Land(EnGoma* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelanime);
+
+    if ((this->actor.bgCheckFlags & 1) != 0) {
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:478]:
```suggestion
    if (this->actor.bgCheckFlags & 1)  {
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 17]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+        Math_ApproachZeroF(&this->actor.speedXZ, 1.0f, 2.0f);
+    }
+    if (this->actionTimer == 0) {
+        EnGoma_SetupStand(this);
+    }
+}
+
+void EnGoma_SetupJump(EnGoma* this) {
+    Animation_Change(&this->skelanime, &D_06000544, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000544), 2, 0.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:487]:
```suggestion
    Animation_Change(&this->skelanime, &D_06000544, 1.0f, 0.0f, Animation_GetLastFrame(&D_06000544), ANIMMODE_ONCE, 0.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 18]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    }
+    if (eyeYaw < -6000) {
+        eyeYaw = -6000;
+    }
+    Math_ApproachS(&this->eyeYaw, eyeYaw, 3, 2000);
+    Math_ApproachS(&this->eyePitch, eyePitch, 3, 2000);
+}
+
+void EnGoma_UpdateHit(EnGoma* this, GlobalContext* globalCtx) {
+    static Vec3f sShieldKnockbackVel = { 0, 0, 20 };
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:604]:
```suggestion
    static Vec3f sShieldKnockbackVel = { 0.0f, 0.0f, 20.0f };
```
Unless it needed to be the other way.

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 19]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+
+        if ((this->colCyl1.base.atFlags & 2) && this->actionFunc == EnGoma_Jump) {
+            EnGoma_SetupLand(this);
+            this->actor.speedXZ = 0.0f;
+            this->actor.velocity.y = 0.0f;
+        }
+
+        if ((this->colCyl2.base.acFlags & 2) && (s8)this->actor.colChkInfo.health > 0) {
+            acHitInfo = this->colCyl2.info.acHitInfo;
+            this->colCyl2.base.acFlags &= ~2;
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:621]:
```suggestion
        if ((this->colCyl2.base.acFlags & AC_HIT) && (s8)this->actor.colChkInfo.health > 0) {
            acHitInfo = this->colCyl2.info.acHitInfo;
            this->colCyl2.base.acFlags &= ~AC_HIT;
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 20]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+    if (this->invincibilityTimer != 0) {
+        this->invincibilityTimer--;
+    }
+
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(thisx);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:707]:
```suggestion
    Actor_MoveForward(&this->actor);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 21]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+
+    if (this->actor.params < 10) {
+        this->eggTimer++;
+        Math_SmoothStepToF(&this->actor.scale.x, 0.01f, 0.5f, 0.00075f, 0.000001f);
+        Math_SmoothStepToF(&this->actor.scale.y, 0.01f, 0.5f, 0.00075f, 0.000001f);
+        Math_SmoothStepToF(&this->actor.scale.z, 0.01f, 0.5f, 0.00075f, 0.000001f);
+        EnGoma_UpdateHit(this, globalCtx);
+        Actor_UpdateBgCheckInfo(globalCtx, thisx, 50.0f, 50.0f, 100.0f, 5);
+        EnGoma_SetFloorRot(this);
+        Actor_SetFocus(thisx, 20.0f);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:721]:
```suggestion
        Actor_SetFocus(&this->actor, 20.0f);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 22]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+            this->colCyl2.dim.height = 35;
+            this->colCyl2.dim.yShift = 0;
+        } else {
+            this->colCyl2.dim.radius = 15;
+            this->colCyl2.dim.height = 30;
+            this->colCyl2.dim.yShift = 10;
+        }
+        if (this->invincibilityTimer == 0) {
+            Collider_UpdateCylinder(thisx, &this->colCyl1);
+            Collider_UpdateCylinder(thisx, &this->colCyl2);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:736]:
```suggestion
            Collider_UpdateCylinder(&this->actor, &this->colCyl1);
            Collider_UpdateCylinder(&this->actor, &this->colCyl2);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 23]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+                       (s16)(Rand_ZeroOne() * 255.0f), 255);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_goma.c", 2011);
+    return 0;
+}
+
+Gfx* EnGoma_NoBackfaceCullingDlist(GraphicsContext* gfxCtx) {
+    Gfx* dl;
+    Gfx* dlStart;
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:765]:
```suggestion
    Gfx* dList;
    Gfx* dListHead;
```
Is what we usually call these.

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 24]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+            Matrix_Translate(this->actor.world.pos.x,
+                             this->actor.world.pos.y + ((this->actor.shape.yOffset * this->actor.scale.y) +
+                                                        globalCtx->mainCamera.skyboxOffset.y),
+                             this->actor.world.pos.z, 0);
+            Matrix_RotateX(this->slopePitch / 32768.0f * 3.1415927f, 1);
+            Matrix_RotateZ(this->slopeRoll / 32768.0f * 3.1415927f, 1);
+            Matrix_RotateY(this->actor.shape.rot.y / 32768.0f * 3.1415927f, 1);
+            Matrix_RotateX(this->actor.shape.rot.x / 32768.0f * 3.1415927f, 1);
+            Matrix_RotateZ(this->actor.shape.rot.z / 32768.0f * 3.1415927f, 1);
+            Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, 1);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:795]:
```suggestion
            Matrix_RotateX(this->slopePitch / 32768.0f * 3.1415927f, MTXMODE_APPLY);
            Matrix_RotateZ(this->slopeRoll / 32768.0f * 3.1415927f, MTXMODE_APPLY);
            Matrix_RotateY(this->actor.shape.rot.y / 32768.0f * 3.1415927f, MTXMODE_APPLY);
            Matrix_RotateX(this->actor.shape.rot.x / 32768.0f * 3.1415927f, MTXMODE_APPLY);
            Matrix_RotateZ(this->actor.shape.rot.z / 32768.0f * 3.1415927f, MTXMODE_APPLY);
            Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 25]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+            Matrix_Push();
+            Matrix_Scale(this->eggScale, 1.0f / this->eggScale, this->eggScale, 1);
+            Matrix_RotateY(this->eggSquishAngle * 0.15f, 1);
+            Matrix_RotateZ(this->eggSquishAngle * 0.1f, 1);
+            Matrix_Scale(0.95f - this->eggSquishAmount, this->eggSquishAmount + 1.05f, 0.95f - this->eggSquishAmount,
+                         1);
+            Matrix_RotateZ(-(this->eggSquishAngle * 0.1f), 1);
+            Matrix_RotateY(-(this->eggSquishAngle * 0.15f), 1);
+            Matrix_Translate(0.0f, this->eggYOffset, 0.0f, 1);
+            Matrix_RotateX(this->eggPitch, 1);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:814]:
```suggestion
            Matrix_Scale(this->eggScale, 1.0f / this->eggScale, this->eggScale, MTXMODE_APPLY);
            Matrix_RotateY(this->eggSquishAngle * 0.15f, MTXMODE_APPLY);
            Matrix_RotateZ(this->eggSquishAngle * 0.1f, MTXMODE_APPLY);
            Matrix_Scale(0.95f - this->eggSquishAmount, this->eggSquishAmount + 1.05f, 0.95f - this->eggSquishAmount,
                         MTXMODE_APPLY);
            Matrix_RotateZ(-(this->eggSquishAngle * 0.1f), MTXMODE_APPLY);
            Matrix_RotateY(-(this->eggSquishAngle * 0.15f), MTXMODE_APPLY);
            Matrix_Translate(0.0f, this->eggYOffset, 0.0f, MTXMODE_APPLY);
            Matrix_RotateX(this->eggPitch, MTXMODE_APPLY);
```

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 26]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+void EnGoma_Debris(EnGoma* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += 2500;
+    this->actor.shape.rot.x += 3500;
+    if (this->actionTimer == 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnGoma_SpawnHatchDebris(EnGoma* this, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:848]:
```suggestion
void EnGoma_SpawnHatchDebris(EnGoma* this, GlobalContext* globalCtx2) {
    GlobalContext* globalCtx = globalCtx2;
```
And switch them in the functions that use them.

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 27]
```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    }
+
+    if (this->actionTimer < 121) {
+        if (Math_SmoothStepToF(&this->actor.scale.y, 0.0f, 1.0f, 0.00075f, 0) <= 0.001f) {
+            Actor_Kill(&this->actor);
+        }
+        this->actor.scale.x = this->actor.scale.z = this->actor.scale.y;
+    }
+
+    if (((this->actionTimer & 7) == 0) && (this->actionTimer != 0)) {
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:893]: Can this be `%`?

**Review by louist103 — CHANGES_REQUESTED** [review]
Nice job on this.  Just some style things.

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c (hunk 7 above)
Comment (Synray) [reply src/overlays/actors/ovl_En_Goma/z_en_goma.c:297]: looool yeah forgot about that one

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 28]
```diff
@@ -761,15 +760,15 @@ s32 EnGoma_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList
...
+    Gfx* dList;
+    Gfx* dListHead;
 
-    dl = dlStart = Graph_Alloc(gfxCtx, 32);
-    gDPPipeSync(dl++);
-    gDPSetRenderMode(dl++, G_RM_PASS, G_RM_AA_ZB_TEX_EDGE2);
-    gSPClearGeometryMode(dl++, G_CULL_BACK);
-    gSPEndDisplayList(dl++);
-    return dlStart;
+    dList = dListHead = Graph_Alloc(gfxCtx, 32);
```
Comment (louist103) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:766]: sizeof

**Review by louist103 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 29]
```diff
@@ -73,77 +104,795 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    if (!(this->eggTimer & 0xF) && Rand_ZeroOne() < 0.5f) {
+        for (i = 0; i < 2; i++) {
+            Vec3f vel = { 0.0f, 0.0f, 0.0f };
+            Vec3f acc = { 0.0f, -0.5f, 0.0f };
+            Vec3f pos;
+
+            pos.x = Rand_CenteredFloat(30.0f) + this->actor.world.pos.x;
+            pos.y = Rand_ZeroFloat(30.0f) + this->actor.world.pos.y;
+            pos.z = Rand_CenteredFloat(30.0f) + this->actor.world.pos.z;
+            EffectSsHahen_Spawn(globalCtx, &pos, &vel, &acc, 0, (s16)(Rand_ZeroOne() * 5.0f) + 10, -1, 10, 0);
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:300]: last arg on hahen is NULL, and you can include hahens header and use the define for -1

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 30]
```diff
@@ -73,77 +104,795 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    }
+
+    if (this->actionTimer == 0 && Math_SmoothStepToF(&this->actor.scale.y, 0.0f, 0.5f, 0.00225f, 0.00001f) <= 0.001f) {
+        if (this->actor.params < 6) {
+            BossGoma* parent = (BossGoma*)this->actor.parent;
+            parent->childrenGohmaState[this->actor.params] = -1;
+        }
+        Audio_PlaySoundGeneral(NA_SE_EN_EXTINCT, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        Actor_Kill(&this->actor);
+        Item_DropCollectibleRandom(globalCtx, NULL, &this->actor.world.pos, 48);
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:425]: would leave the last arg in hex for now, its some kind of flags

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 31]
```diff
@@ -73,77 +104,795 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    this->visualState = 2;
+}
+
+void EnGoma_SetupStand(EnGoma* this) {
+    f32 lastFrame;
+
+    lastFrame = Animation_GetLastFrame(&D_06001548);
+    this->actionTimer = Rand_S16Offset(10, 30);
+    Animation_Change(&this->skelanime, &D_06001548, 1.0f, 0.0f, lastFrame, ANIMMODE_LOOP, -5.0f);
+    this->actionFunc = &EnGoma_Stand;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:436]: can remove & on function pointers, if there are anymore in the file as well

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 32]
```diff
@@ -73,77 +104,795 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+                        swordDamage = 1;
+                    }
+
+                    this->actor.colChkInfo.health -= swordDamage;
+                    EnGoma_SetupHurt(this, globalCtx);
+                    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 5);
+                    this->hurtTimer = 13;
+                }
+            } else {                           // die if still an egg
+                if (this->actor.params <= 5) { //! BossGoma only has 3 children
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:655]: think moving the comment down one line into the else block will play nicer with the formatter

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 33]
```diff
@@ -73,77 +104,795 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+                                                        globalCtx->mainCamera.skyboxOffset.y),
+                             this->actor.world.pos.z, MTXMODE_NEW);
+            Matrix_RotateX(this->slopePitch / 32768.0f * 3.1415927f, MTXMODE_APPLY);
+            Matrix_RotateZ(this->slopeRoll / 32768.0f * 3.1415927f, MTXMODE_APPLY);
+            Matrix_RotateY(this->actor.shape.rot.y / 32768.0f * 3.1415927f, MTXMODE_APPLY);
+            Matrix_RotateX(this->actor.shape.rot.x / 32768.0f * 3.1415927f, MTXMODE_APPLY);
+            Matrix_RotateZ(this->actor.shape.rot.z / 32768.0f * 3.1415927f, MTXMODE_APPLY);
+            Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+            SkelAnime_DrawOpa(globalCtx, this->skelanime.skeleton, this->skelanime.jointTable, &EnGoma_OverrideLimbDraw,
+                              0, this);
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:796]: second to last arg can be NULL

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 34]
```diff
@@ -73,77 +105,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_goma.c", 2011);
+    return 0;
+}
+
+Gfx* EnGoma_NoBackfaceCullingDlist(GraphicsContext* gfxCtx) {
+    Gfx* dList;
+    Gfx* dListHead;
+
+    dList = dListHead = Graph_Alloc(gfxCtx, sizeof(Gfx) * 4);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:768]: It should be the other way around (`dList` and `dListHead` being incremented)

File: src/overlays/actors/ovl_En_Goma/z_en_goma.c [hunk 35]
```diff
@@ -73,77 +105,796 @@ static ColliderCylinderInit D_80A4B7CC = {
...
+    if (!(this->eggTimer & 0xF) && Rand_ZeroOne() < 0.5f) {
+        for (i = 0; i < 2; i++) {
+            Vec3f vel = { 0.0f, 0.0f, 0.0f };
+            Vec3f acc = { 0.0f, -0.5f, 0.0f };
+            Vec3f pos;
+
+            pos.x = Rand_CenteredFloat(30.0f) + this->actor.world.pos.x;
+            pos.y = Rand_ZeroFloat(30.0f) + this->actor.world.pos.y;
+            pos.z = Rand_CenteredFloat(30.0f) + this->actor.world.pos.z;
+            EffectSsHahen_Spawn(globalCtx, &pos, &vel, &acc, 0, (s16)(Rand_ZeroOne() * 5.0f) + 10, HAHEN_OBJECT_DEFAULT, 10, NULL);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Goma/z_en_goma.c:301]: This looks like it needs to be formatted one last time

**Comment by fig02** [comment]
changes made

**Review by Roman971 — APPROVED** [review]

**Pull request merged**